README.md

```diff
@@ -1,2 +1,18 @@
 # evm-bridge-contracts
 Contracts to support the trustless bridge of the EOS EVM
+## Dependency
+- cmake 3.16 or later
+- install cdt
+- solc: Used to compile the .sol file. We choose to use solcjs because it is more actively maintained then the solc from rpm.
+    - Install node.js and npm
+    - Install solcjs: 'npm install -g solc'
+
+## How to Build
+- cd to 'build' directory
+- run the command 'cmake -Deosio_DIR={leap dir} ..'
+- run the command 'make'
```

Comment:
If you make the CMakeLists.txt changes to follow how things are done in eos-system-contracts as opposed to the undesirable way we do things currently in eos-evm, then we we should be able to build all contracts and tests even with custom built CDT and Leap using the instructions in the "Build system contracts with tests using Leap and CDT both built from source" section of the eos-system-contracts README:

```shell
mkdir build
cd build
cmake -DCMAKE_BUILD_TYPE=Release -Dcdt_DIR="${CDT_BUILD_PATH}/lib/cmake/cdt" -Dleap_DIR="${LEAP_BUILD_PATH}/lib/cmake/leap" ..
make -j $(nproc)
```

Note: We do not need to bother following the complicated version checks in the CMakeLists.txt for the tests as is done in eos-system-contracts. We can also always force the tests to be built by default. We added the `BUILD_TESTS` option in eos-system-contracts so that people could build the contracts without needing to worry about the Leap dependency. But we can just require the Leap dependency for anyone trying to build these contracts. 

That means you should update the dependencies list above to also include Leap 4.0 or later. Also add a minimum version of 4.0 for CDT in the dependency list.

Reply:
updated readme